### PR TITLE
Add `sessionId` and `integrationType` to `BraintreeClient` constructor

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Add `sessionId` and `integrationType` to `BraintreeClient` constructor to allow them to be passed from `DropInClient`
 - `sessionId` allows Drop-in to maintain the same session ID for analytics events during a Drop-in payment flow
 - `integrationType` was previously based on asserting on a `DropInActivity` class - now that the merchant owns the launching activity, this no longer works in Drop-in, updating to make IntegrationType an enum and allow Drop-in to pass type "dropin"

 ### Checklist

 - ~[ ] Added a changelog entry~

- @sarahkoop

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]
🌊  🚀

File: BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.java [hunk 1]
```diff
@@ -35,17 +36,30 @@ private static BraintreeClientParams createDefaultParams(Context context, String
...
                 .replace("_", "") + ".braintree";
-        return createDefaultParams(context, authString, returnUrlScheme);
+        return createDefaultParams(context, authString, returnUrlScheme, UUIDHelper.getFormattedUUID(), IntegrationType.CUSTOM);
     }
 
     private static BraintreeClientParams createDefaultParams(Context context, String authString, String returnUrlScheme) {
+        return createDefaultParams(context, authString, returnUrlScheme, UUIDHelper.getFormattedUUID(), IntegrationType.CUSTOM);
+    }
+
+    private static BraintreeClientParams createDefaultParams(Context context, String authString, String sessionId, @IntegrationType.Integration String integrationType) {
```
Comment (scannillo) [inline BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.java:46]: Maybe a docstring or note clarifying that this is meant for DropIn to enabled proper analytics.

Comment (scannillo) [reply BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.java:46]: We could even rename the func to be more clear that is it's purpose.

**Review by scannillo — APPROVED** [review]
Overall looks good! Left one small comment.

**Pull request merged**